### PR TITLE
Fixes for the Python 3 move, file input revamps and spark job runs.

### PR DESCRIPTION
**Pull request by jezdez**
This is the result of QA I did on the latest tag https://github.com/mozilla/telemetry-analysis-service/tree/2017.2.8

### CODE REVIEW

File: atmo/jobs/forms.py [hunk 1]
```diff
@@ -180,6 +172,17 @@ def save(self, commit=True):
...
+        super().__init__(*args, **kwargs)
+        self.fields['identifier'].widget.attrs.update({
+            'data-parsley-remote': (
+                reverse('jobs-identifier-available') + '?identifier={value}'
+            ),
+            'data-parsley-remote-reverse': 'true',
+            'data-parsley-remote-message': 'Identifier unavailable',
+            'data-parsley-debounce': '500',
+        })
+
```
Comment (jezdez) [inline atmo/jobs/forms.py]: This basically prevents saving job during editing.

File: atmo/jobs/migrations/0012_add_job_run_history.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='sparkjobrun',
             name='spark_job',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='run', related_query_name=b'runs', to='jobs.SparkJob'),
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='runs', related_query_name='runs', to='jobs.SparkJob'),
         ),
```
Comment (jezdez) [inline atmo/jobs/migrations/0012_add_job_run_history.py]: Some Python 3 and one naming fix.

File: atmo/jobs/views.py [hunk 3]
```diff
@@ -120,7 +120,7 @@ def detail_spark_job(request, id):
 def download_spark_job(request, id):
     spark_job = SparkJob.objects.get(pk=id)
     response = StreamingHttpResponse(
-        spark_job.notebook_s3_object['Body'].read(),
+        spark_job.notebook_s3_object['Body'].read().decode('utf-8'),
         content_type='application/x-ipynb+json',
```
Comment (jezdez) [inline atmo/jobs/views.py]: Without decoding the bytestream the streaming response would not be correctly rendered

File: atmo/templates/atmo/base.html [hunk 4]
```diff
@@ -97,6 +97,7 @@
     <script type="text/javascript" src="{% static "npm/bootstrap-datetime-picker/js/bootstrap-datetimepicker.min.js" %}"></script>
     <script type="text/javascript" src="{% static "npm/parsleyjs/dist/parsley.min.js" %}"></script>
     <script type="text/javascript" src="{% static "js/base.js" %}"></script>
+    <script type="text/javascript" src="{% static "js/fileinput.js" %}"></script>
```
Comment (jezdez) [inline atmo/templates/atmo/base.html]: This was kind of a huge omission when I moved the jasny stuff in f47ab9b3f735efd7925962d2d22f02bcaccddab9. My bad :(

File: atmo/jobs/migrations/0012_add_job_run_history.py (hunk 2 above)
Comment (robhudson) [inline atmo/jobs/migrations/0012_add_job_run_history.py]: Oops. Sorry. I think I did change it after creating the migration.

**Pull request merged**